File: bffGrocery/src/main/java/com/gitlab/arsenji/bffGrocery/dto/ProductDto.java
package com.gitlab.arsenji.bffGrocery.dto;

import com.fasterxml.jackson.annotation.JsonProperty;

public record ProductDto(
        @JsonProperty("rails_id") Long railsId,
        String barcode,
        @JsonProperty("product_name") String productName,
        Integer quantity,
        @JsonProperty("measurement_unit") String measurementUnit
){}

File: mobile/app/src/main/java/com/github/arseniji/barcodescanner/network/dto/ProductDto.kt
package com.github.arseniji.barcodescanner.network.dto

import com.google.gson.annotations.SerializedName

data class ProductDto(
    @SerializedName("rails_id") val railsId: Int,
    val barcode: String,
    @SerializedName("product_name") val productName: String,
    val quantity: Int,
    @SerializedName("measurement_unit") val measurementUnit: String?,
)